src/task/MonitorTask.ts

```diff
@@ -296,11 +298,41 @@ export default class MonitorTask
             Maybe.of(observer.connectionDidBecomePoor).map(f => f.bind(observer)());
           });
         }
-      } else {
+      },
+      () => {
         this.context.audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
           Maybe.of(observer.connectionDidBecomeGood).map(f => f.bind(observer)());
         });
       }
+    );
+
+    this.enactHealthPolicy(
+      this.sendingAudioFailureHealthPolicy,
+      connectionHealthData,
+      () => {
+        this.context.eventController?.publishEvent('sendingAudioFailed');
+      },
+      () => {
+        this.context.eventController?.publishEvent('sendingAudioRecovered');
+      }
+    );
+  }
+
+  private enactHealthPolicy(
+    healthPolicy: BaseConnectionHealthPolicy,
+    connectionHealthData: ConnectionHealthData,
+    unHealthyCallback: () => void,
+    healthyCallback?: () => void
+  ): void {
+    healthPolicy.update(connectionHealthData);
+    const healthValue = healthPolicy.healthIfChanged();
+    if (healthValue !== null) {
+      this.logger.info(`${healthPolicy.getCanonicalNameForLogging()} value is now ${healthValue}`);
+      if (healthValue <= healthPolicy.minimumHealth()) {
+        unHealthyCallback.bind(this)();
+      } else {
+        Maybe.of(healthyCallback).map(f => f.bind(this)());
```

Comment:
`healthyCallback` also does not need to map on observers since it is not an array? Sorry correct me if wrong.

Reply:
Sorry, I don't understand your comment. Based on my understanding `Maybe` can be used with any function to safely invoke after checking for null/undefined. But I don't really need it here because I'm not using the return value after invocation anyways and not chaining any calls.  I can avoid using Maybe by doing the following:

```
if (healthyCallback) {
  healthyCallback.bind(this)();
}
```

Is that what you're suggesting?

Reply:
Sorry I was not clear earlier. But you answered it and thanks for explanation. I was unaware on the `Maybe.of` returning a [`map`](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/utils/Types.ts#L38). 

If you are making both callbacks optional then its needed for both now. Elsewhere, we follow the same so its fine to use here as well.